internal/metamorph/processor.go

```diff
@@ -384,6 +394,37 @@ func (p *Processor) StartSendStatusUpdate() {
 	}()
 }
 
+func (p *Processor) StartCheckingTransactionsInNetwork() {
+	p.waitGroup.Add(1)
+	ticker := time.NewTicker(100 * time.Millisecond)
+	go func() {
+		defer p.waitGroup.Done()
+		for {
+			select {
+			case <-p.ctx.Done():
+				return
+
+			case <-ticker.C:
+				p.announcedTransactionsLock.Lock()
+				for k := 0; k < len(p.announcedTransactions); k++ {
+					if p.announcedTransactions[k].second < uint64(time.Now().Unix())-1 {
```

Comment:
1. use `p.now()`
2. move calculating timestamp (`uint64(time.Now().Unix())-1`) above the loop
3. remove redundant casting from `p.pm.RequestTransaction((*chainhash.Hash)(p.announcedTransactions[k].hash))`

Reply:
Don't understand second point here.

Reply:
```
for k := 0; k < len(p.announcedTransactions); k++ {
	if p.announcedTransactions[k].second < uint64(time.Now().Unix())-1{
	      (...)
	}
	(...)
}
					
```
To:
```
until :=  uint64(p.Now().Unix())-1
for k := 0; k < len(p.announcedTransactions); k++ {
	if p.announcedTransactions[k].second < until{
	    (...)
	}
	(...)
}